woocommerce/woocommerce-ios
# Coupons: Populate coupons from local storage upon immediately during init stage

### Description

There's an issue in the coupon list that we've been missing: the coupon list is not populating data from local storage on initial load. We're already fetching local data but not using them immediately; the list is only updated when there's a change or reset to the data. This leads to an empty list when opening the screen without an internet connection, and the list always shows the loading state on the first load even though there are data cached locally.

The solution is one simple line to populate the data right after the first data fetch.

### Testing instructions

Precondition: please make sure that your store has Coupons enabled (WP Admin > WC Settings > General > Enable Coupons) and has set up at least one coupon.
- Launch the app, make sure to enable Coupon Management in Experimental Features.
- With the internet connection on, navigate to Menu > Coupons to fetch the list if you haven't done already.
- Navigate back out, turn off the connection. Navigate back to Menu > Coupons, notice that the previously fetched item has been populated already.
- Repeat the last step with the connection back on. Notice that the list displays data immediately without showing the loading animation.

### Screenshots

Before:
https://user-images.githubusercontent.com/5533851/159199333-4e4378f4-0758-4e5f-991e-490bbf2dd955.mov

After:
https://user-images.githubusercontent.com/5533851/159199369-0c32573f-7f1a-4661-a0c2-f7101bb91be5.MP4


---
